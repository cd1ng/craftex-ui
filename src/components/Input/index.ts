export { Input } from './Input'
export type { InputProps,InputOnChangeData } from './types'