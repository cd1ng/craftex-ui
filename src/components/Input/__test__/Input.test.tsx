import { render } from "@testing-library/react";
import { Input } from "../Input";

describe("Input",()=>{
  it("renders a default input",()=>{
    const component = render(<Input />);
    // 返回一个渲染组件的DocumentFragment
    // 使用 snapshot 时，Vitest 将拍摄给定值的快照，然后将其与与测试一起存储的参考快照文件进行比较。如果两个快照不匹配，则测试将失败：更改是意外的，或者引用快照需要更新到结果的新版本。
    const tree = component.asFragment();
    expect(tree).toMatchSnapshot();
  })

  it("renders a small input",()=>{
    const smallComponent = render(<Input size="small" />);
    const tree = smallComponent.asFragment();
    expect(tree).toMatchSnapshot();
  })

  it("renders a medium input",()=>{
    const mediumComponent = render(<Input size="medium" />);
    const tree = mediumComponent.asFragment();
    expect(tree).toMatchSnapshot();
  })

  it("renders a large input",()=>{
    const largeComponent = render(<Input size="large" />);
    const tree = largeComponent.asFragment();
    expect(tree).toMatchSnapshot();
  })

  it("renders a disabled input",()=>{
    const disabledComponent = render(<Input disabled />);
    const tree = disabledComponent.asFragment();
    expect(tree).toMatchSnapshot();
  })

})