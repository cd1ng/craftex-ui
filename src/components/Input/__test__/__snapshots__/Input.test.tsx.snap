// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input > renders a default input 1`] = `
<DocumentFragment>
  <span
    class="fish-ui-Input r1oeeo9n ___1v9icnz_137yv9i fvcxoqz f1ub3y4t f1l4zc64 f1m52nbi f8vnjqi fz1etlk f1klwx88 f1hc16gm"
  >
    <input
      class="fish-ui-Input__input r12stul0"
      type="text"
      value=""
    />
  </span>
</DocumentFragment>
`;

exports[`Input > renders a disabled input 1`] = `
<DocumentFragment>
  <span
    class="fish-ui-Input r1oeeo9n ___l8rfd20_fexo0v0 fdrzuqr f1c21dwh f1jj8ep1 f15xbau f4ikngz fy0fskl f9dbb4x f3qs60o fwd1oij f5u9ap2 fhr9occ f99w1ws"
  >
    <input
      class="fish-ui-Input__input r12stul0 ___12qx8ed_15h66xx f1s2aq7o f1c21dwh fdrzuqr fahhnxm"
      type="text"
      value=""
    />
  </span>
</DocumentFragment>
`;

exports[`Input > renders a large input 1`] = `
<DocumentFragment>
  <span
    class="fish-ui-Input r1oeeo9n ___1od6sq6_1wz9i3l f1w5jphr fk6fouc fod5ikn figsok6 faaz57k f1eyhf9v fvcxoqz f1ub3y4t f1l4zc64 f1m52nbi f8vnjqi fz1etlk f1klwx88 f1hc16gm"
  >
    <input
      class="fish-ui-Input__input r12stul0 ___e4pcmp0_u8glci0 fnphzt9 flt1dlf"
      type="text"
      value=""
    />
  </span>
</DocumentFragment>
`;

exports[`Input > renders a medium input 1`] = `
<DocumentFragment>
  <span
    class="fish-ui-Input r1oeeo9n ___1v9icnz_137yv9i fvcxoqz f1ub3y4t f1l4zc64 f1m52nbi f8vnjqi fz1etlk f1klwx88 f1hc16gm"
  >
    <input
      class="fish-ui-Input__input r12stul0"
      type="text"
      value=""
    />
  </span>
</DocumentFragment>
`;

exports[`Input > renders a small input 1`] = `
<DocumentFragment>
  <span
    class="fish-ui-Input r1oeeo9n ___1hq2tbx_1ql12s0 f1pha7fy fk6fouc fy9rknc figsok6 fwrc4pm fvcxoqz f1ub3y4t f1l4zc64 f1m52nbi f8vnjqi fz1etlk f1klwx88 f1hc16gm"
  >
    <input
      class="fish-ui-Input__input r12stul0 ___dmmac90_19wzd1i f1f5gg8d f1vdfbxk"
      type="text"
      value=""
    />
  </span>
</DocumentFragment>
`;
